src/linux/x86_64/kvm_cpu.rs

```diff
@@ -75,34 +104,13 @@ impl VirtualizationBackendInternal for KvmVm {
 	) -> HypervisorResult<Self> {
 		let vm = KVM.create_vm().unwrap();
 
-		let sz = std::cmp::min(peripherals.mem.memory_size, KVM_32BIT_GAP_START);
-
-		let kvm_mem = kvm_userspace_memory_region {
-			slot: 0,
-			flags: peripherals.mem.flags,
-			memory_size: sz as u64,
-			guest_phys_addr: peripherals.mem.guest_address.as_u64(),
-			userspace_addr: peripherals.mem.host_address as u64,
-		};
-
-		unsafe { vm.set_user_memory_region(kvm_mem) }?;
-
-		if peripherals.mem.memory_size > KVM_32BIT_GAP_START + KVM_32BIT_GAP_SIZE {
-			let kvm_mem = kvm_userspace_memory_region {
-				slot: 1,
-				flags: peripherals.mem.flags,
-				memory_size: (peripherals.mem.memory_size
-					- KVM_32BIT_GAP_START
-					- KVM_32BIT_GAP_SIZE) as u64,
-				guest_phys_addr: peripherals.mem.guest_address.as_u64()
-					+ (KVM_32BIT_GAP_START + KVM_32BIT_GAP_SIZE) as u64,
-				userspace_addr: (peripherals.mem.host_address as usize
-					+ KVM_32BIT_GAP_START
-					+ KVM_32BIT_GAP_SIZE) as u64,
-			};
-
-			unsafe { vm.set_user_memory_region(kvm_mem) }?;
-		}
+		let regions = Vec::<kvm_userspace_memory_region>::from(&peripherals.mem);
+		let _: Vec<_> = regions
+			.iter()
+			.map(|kvm_mem| {
```

Comment:
```suggestion
			.try_for_each(|kvm_mem| {
```

and get rid of the `.collect()` below.